### PR TITLE
Description about integration of Metrics with Logging

### PR DESCRIPTION
**Pull request by jpkrohling**

### CODE REVIEW

**Comment by jpkrohling** [comment]
This depends on openshift/origin-metrics#347 , but this is ready to be reviewed.

File: install_config/cluster_metrics.adoc [hunk 1]
```diff
@@ -720,6 +720,32 @@ Information about scaling cluster metrics capabilities is available in the
 xref:../scaling_performance/scaling_cluster_metrics.adoc#cluster-metrics-scaling-openshift-metrics-pods[Scaling and
 Performance Guide].
 
+[[metrics-logging]]
+== Integration with Aggregated Logging
+
+Hawkular Alerts might be connected to the Aggregated Logging's Elasticsearch to react on log events. By default,
```
Comment (ahardin-rh) [inline install_config/cluster_metrics.adoc]: s/might/must

Comment (ahardin-rh) [inline install_config/cluster_metrics.adoc]: what is meant by "react on log events"?

File: install_config/cluster_metrics.adoc [hunk 2]
```diff
@@ -720,6 +720,32 @@ Information about scaling cluster metrics capabilities is available in the
...
+Hawkular will try to find Elasticsearch on its default place (namespace `logging`, pod `logging-es`) at every boot.
+If the Aggregated Logging is installed after Hawkular, the Hawkular Metrics pod might need to be restarted in order
+to recognize the new Elasticsearch server. The Hawkular boot log provides a clear indication if the integration could
+not be properly configured, with messages like:
+
+> Failed to import the logging certificate into the store. Continuing, but the logging integration might fail.
```
Comment (ahardin-rh) [inline install_config/cluster_metrics.adoc]: We can place these error messages in code blocks using ----. For example:

     ----
    > Failed to import the logging certificate into the store. Continuing, but the logging integration might fail.
     ----

File: install_config/cluster_metrics.adoc [hunk 3]
```diff
@@ -720,6 +720,32 @@ Information about scaling cluster metrics capabilities is available in the
...
+> Could not get the logging secret! Status code: 000. The Hawkular Alerts integration with Logging might not work properly.
+
+This feature is available from
+ifdef::openshift-origin[]
+ version v1.6,
+endif::[]
+ifdef::openshift-enterprise[]
+ version 3.6.0,
+endif::[]
+and can be positively confirmed to be available by checking a log entry similar to this:
```
Comment (ahardin-rh) [inline install_config/cluster_metrics.adoc]: You can confirm if logging is available by checking the log for an entry like:

File: install_config/cluster_metrics.adoc (hunk 1 above)
Comment (jpkrohling) [inline install_config/cluster_metrics.adoc]: > what is meant by "react on log events"?

Elasticsearch creates log entries (log events), which are later processed by Hawkular Alerts. As outcome of this processing, Hawkular Alerts might trigger a new alert if it detects that it's an `ERROR`.

File: install_config/cluster_metrics.adoc (hunk 3 above)
Comment (jpkrohling) [inline install_config/cluster_metrics.adoc]: Looks better, but just wanted to note that this is part of the same phrase from the previous lines. With the change, this is how it looks like on screen:

> This feature is available from version v1.6. You can confirm if logging is available by checking the log for an entry like:

**Comment by jpkrohling** [comment]
By the way: I'm not sure about the version it's going to be available. The Origin Metrics image for 3.6 has the required changes, but the Ansible Playbook with that feature will not have this for 3.6.

Should I then change the documentation to say that it's available from 3.7 because of the playbook, or having the support in the underlying image is enough to warrant the documentation change for 3.6?

cc @mwringe , @jcantrill , @ewolinetz

**Comment by jcantrill** [comment]
@jpkrohling I would say unless you know for certain the ansible changes will merge by Friday, Jun. 23 this feature should be listed as available for 3.7

**Comment by jpkrohling** [comment]
Looks like I can't change the label on this issue, but I added a commit bumping the version to 1.7/3.7.

**Comment by ahardin-rh** [comment]
@jpkrohling @jcantrill Thank you! I updated the label to 3.7.  Can you please squash the changes into one commit? Once that is done, I can merge. Thanks again!

**Comment by jpkrohling** [comment]
Commits squashed.

**Comment by ahardin-rh** [comment]
[rev_history]
|xref:../install_config/cluster_metrics.adoc#install-config-cluster-metrics[Enabling Cluster Metrics]
|Added a new xref:../install_config/cluster_metrics.adoc#metrics-logging[Integration with Aggregated Logging] section
%

**Pull request merged**